community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/pipes/CountStoreNodeAggregationPipe.scala

```diff
@@ -0,0 +1,41 @@
+package org.neo4j.cypher.internal.compiler.v2_3.pipes
+
+import org.neo4j.cypher.internal.compiler.v2_3.ExecutionContext
+import org.neo4j.cypher.internal.compiler.v2_3.executionplan.{Effects, ReadsNodes}
+import org.neo4j.cypher.internal.compiler.v2_3.planDescription.{NoChildren, PlanDescriptionImpl}
+import org.neo4j.cypher.internal.compiler.v2_3.symbols.SymbolTable
+import org.neo4j.cypher.internal.frontend.v2_3.NameId
+import org.neo4j.cypher.internal.frontend.v2_3.symbols._
+
+case class CountStoreNodeAggregationPipe(ident: String, labelName: Option[String])(val estimatedCardinality: Option[Double] = None)
```

Comment:
Couldn't this take a LazyLabel instead? In that way we don't need to lookup the label id for each row.